app/screens/edit_post/edit_post.tsx

```diff
@@ -0,0 +1,241 @@
+};
+
+type EditPostProps = {
+    componentId: string;
+    post: PostModel;
+    maxPostSize: number;
+}
+
+const EditPost = ({componentId, maxPostSize, post}: EditPostProps) => {
+    const [keyboardType, setKeyboardType] = useState<KeyboardType>('default');
+    const [postMessage, setPostMessage] = useState(post.message);
+    const [cursorPosition, setCursorPosition] = useState(0);
+    const [errorLine, setErrorLine] = useState<string | undefined>();
+    const [errorExtra, setErrorExtra] = useState<string | undefined>();
+    const [rightButtonEnabled, setRightButtonEnabled] = useState(true);
+    const [isUpdating, setIsUpdating] = useState(false);
+    const [postInputTop, setPostInputTop] = useState(0);
+
+    const postInputRef = useRef<PostInputRef>(null);
+    const theme = useTheme();
+    const intl = useIntl();
+    const serverUrl = useServerUrl();
+
+    const styles = getStyleSheet(theme);
+    const closeButtonIcon = useMemo(() => CompassIcon.getImageSourceSync('close', 24, theme.sidebarHeaderTextColor), [theme.sidebarHeaderTextColor]);
+
+    useEffect(() => {
+        setButtons(componentId, {
+            leftButtons: [{
+                icon: closeButtonIcon,
+                ...LEFT_BUTTON,
+            }],
+            rightButtons: [{
+                color: theme.sidebarHeaderTextColor,
+                text: intl.formatMessage({id: 'edit_post.save', defaultMessage: 'Save'}),
+                ...RIGHT_BUTTON,
+            }],
+        });
+    }, [theme.sidebarHeaderTextColor]);
+
+    useEffect(() => {
+        const unsubscribe = Navigation.events().registerComponentListener({
+            navigationButtonPressed: ({buttonId}: { buttonId: string }) => {
+                switch (buttonId) {
+                    case LEFT_BUTTON.id: {
+                        onClose();
+                        break;
+                    }
+                    case RIGHT_BUTTON.id:
+                        onSaveEditedPost();
+                        break;
+                }
+            },
+        }, componentId);
+
+        return () => {
+            unsubscribe.remove();
+        };
+    }, [postMessage]);
+
+    useDidUpdate(() => {
+        // Workaround to avoid iOS emdash autocorrect in Code Blocks
+        if (Platform.OS === 'ios') {
+            onPostSelectionChange();
+        }
+    }, [postMessage]);
+
+    const onPostSelectionChange = useCallback((curPos: number = cursorPosition) => {
+        if (Platform.OS === 'ios') {
+            setKeyboardType(switchKeyboardForCodeBlocks(postMessage, curPos));
+        }
+        setCursorPosition(curPos);
+    }, [cursorPosition, postMessage]);
+
+    const updateCanEditPostButton = useCallback((enabled) => {
+        if (rightButtonEnabled !== enabled) {
+            setRightButtonEnabled(enabled);
+            setButtons(componentId, {
+                leftButtons: [{...LEFT_BUTTON, icon: closeButtonIcon}],
+                rightButtons: [{...RIGHT_BUTTON, enabled}],
```

Comment:
Bug to be fixed: Save button is not being enabled on iOS.  Works fine on Android.

Reply:
here you are not setting the text nor the color again and those are not defined in the constant

Reply:
Corrected.